src/Native/CMakeLists.txt

```diff
@@ -130,20 +130,37 @@ if(CMAKE_SYSTEM_NAME STREQUAL SunOS)
     set(CLR_CMAKE_PLATFORM_UNIX 1)
 endif(CMAKE_SYSTEM_NAME STREQUAL SunOS)
 
+function(cfx_unknown_arch)
```

Comment:
I don't think this function is needed since it's only being called from one place (and that place is always !WIN32) unlike in CoreCLR, where it's used in many places. I'd just put the 'message' where you're calling cfx_unknown_arch instead.